### ISSUE TITLE
Reponse examples not working

### ISSUE THREAD

**Issue by jtannas** [rswag/rswag#545]
<sup>Originally posted by **jmoreno-kueski** August 30, 2022</sup>
I've implemented the response examples as they are on the official GitHub repo page however they go beyond a message saying: 
"# Not yet implemented"</div>

@jmoreno-kueski

**Comment by jtannas** [comment]
@jmoreno-kueski
Do you have a link to the code that we can look at to help diagnose the issue?

**Comment by jmoreno-kueski** [comment]
Thanks a lot for the quick reply. The setup should be be something like:

# Gemfile
ruby '2.6.6'
gem 'rswag-api'
gem 'rswag-ui'

group :development, :test do
  gem 'rspec-rails'
  gem 'rswag-specs'
end

*** SPEC File ***

# spec/integration/blogs_spec.rb
describe 'Blogs API' do
  path '/blogs/{blog_id}' do
    get 'Retrieves a blog' do
      response 200, 'blog found' do
        example 'application/json', :example_key, {
          id: 1,
          title: 'Hello world!',
          content: '...'
        }
        run_test!
        example 'application/json', :example_key_2, {
          id: 1,
          title: 'Hello world!',
          content: '...'
        }
        run_test!
      end
    end
  end
end

running **rake swag** gives the following output´:

Generating Swagger docs ...
Swagger doc generated at [some path]/swagger/v1/swagger.yaml

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Blogs API /blogs/{blog_id} get blog found application/json
     # Not yet implemented
     # ./spec/requests/api/v1/blogs_spec.rb:6

  2) Blogs API /blogs/{blog_id} get blog found application/json
     # Not yet implemented
     # ./spec/requests/api/v1/blogs_spec.rb:12


Finished in 0.04641 seconds (files took 9.48 seconds to load)
6 examples, 0 failures, 2 pending

*** YAML FILE ***
---
openapi: 3.0.1
info:
  title: API V1
  version: v1
paths:
  "/blogs/{blog_id}":
    get:
      summary: Retrieves a blog
      responses:
        '200':
          description: blog found

And that's it, it just ignore the example in the output file

**Comment by jtannas** [comment]
What you have here is pretty difficult to follow along...
Could you post a link to a gist or the project github? Or maybe use some of the github code block formatting?

**Comment by romanblanco** [comment]
@jmoreno-kueski I've noticed that you have `run_test!` call in between the `example` calls that is not in the example in [README](https://github.com/rswag/rswag#response-examples).

```ruby
describe 'Blogs API' do
  path '/blogs/{blog_id}' do
    get 'Retrieves a blog' do
      response 200, 'blog found' do
        example 'application/json', :example_key, {
          id: 1,
          title: 'Hello world!',
          content: '...'
        }
        run_test! # <-- extra?
        example 'application/json',:example_key_2, {
          id: 1,
          title: 'Hello world!',
          content: '...'
        }
        run_test!
      end
    end
  end
end
```

I'm not sure why the first example gets ignored in the results, but have you tried removing the middle `run_test!` call?

Like this:

```ruby
describe 'Blogs API' do
  path '/blogs/{blog_id}' do
    get 'Retrieves a blog' do
      response 200, 'blog found' do
        example 'application/json', :example_key, {
            id: 1,
            title: 'Hello world!',
            content: '...'
          }
        example 'application/json', :example_key_2, {
          id: 1,
          title: 'Hello world!',
          content: '...'
        }
        run_test!
      end
    end
  end
end
```

**Comment by jmoreno-kueski** [comment]
Sorry for the delay, I've **removed the middle run_test!**, same result.

**Comment by romanblanco** [comment]
This seems to be duplicate of https://github.com/rswag/rswag/issues/548.

@jmoreno-kueski Looks like using the `example` method is causing issues. Using the legacy `examples` *(plural)* method could temporarily solve your problems until this is resolved:

```ruby
describe 'Blogs API' do
  path '/blogs/{blog_id}' do
    get 'Retrieves a blog' do
      response 200, 'blog found' do
        # FIXME: replace legacy 'examples' method (github.com/rswag/rswag/issues/545) 
        examples 'application/json' => {
            id: 1,
            title: 'Hello world!',
            content: '...'
          }
        examples 'application/json' => {
          id: 1,
          title: 'Hello world!',
          content: '...'
        }
        run_test!
      end
    end
  end
end
```

**Comment by jmoreno-kueski** [comment]
@romanblanco IN my case, the second example overwrites the first example so still one example in the swagger file.
rswag-specs (2.5.1)

      response 200, 'stores found' do
        schema type: :object,
               properties: {
                 status: { type: :string, required: true },
                 code: { type: :string },
                 message: { type: :string },
                 data: { type: :object }
               }

        examples 'application/json' => {
          id: 2,
          title: 'Hello world! asdasdasd',
          content: '...'
        }

        examples 'application/json' => {
          id: 333,
          title: 'Hello world! 5546546',
          content: '...'
        }
        run_test!
      end

**Comment by romanblanco** [comment]
@jmoreno-kueski I see, the legacy method only allows single example, see https://github.com/rswag/rswag/blame/7e0289005218002c39e66a464a9ce82525877226/CHANGELOG.md#L8

**Comment by romanblanco** [comment]
@jmoreno-kueski The new `example` method has been released in version 2.6.0.

I was able to reproduce the issue when using Rswag 2.5.1. 

```
Generating Swagger docs ...
Swagger doc generated at /app/swagger/v1/openapi.json

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Benchmarks API /api/compliance/benchmarks/{id} get retrieves a benchmark application/json
     # Not yet implemented
     # ./spec/integration/benchmarks_spec.rb:126


Finished in 2.42 seconds (files took 1.24 seconds to load)
27 examples, 0 failures, 1 pending
```

After updating to Rswag 2.6.0, the problem stopped.


```diff
diff --git a/Gemfile.lock b/Gemfile.lock
index a6b7a9701..be5cde731 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -326,13 +326,13 @@ GEM
       rspec-mocks (~> 3.9)
       rspec-support (~> 3.9)
     rspec-support (3.9.3)
-    rswag-api (2.5.1)
+    rswag-api (2.6.0)
       railties (>= 3.1, < 7.1)
-    rswag-specs (2.5.0)
+    rswag-specs (2.6.0)
       activesupport (>= 3.1, < 7.1)
       json-schema (~> 2.2)
       railties (>= 3.1, < 7.1)
-    rswag-ui (2.5.1)
+    rswag-ui (2.6.0)
       actionpack (>= 3.1, < 7.1)
       railties (>= 3.1, < 7.1)
     rubocop (1.25.0)
```

```
Generating Swagger docs ...
Swagger doc generated at /app/swagger/v1/openapi.json

Finished in 2.14 seconds (files took 1.27 seconds to load)
26 examples, 0 failures
```

Can you check your `Gemfile.lock` if you're  actually using the new release?

**Comment by jmoreno-kueski** [comment]
@romanblanco You're right man, we just realize about this on Friday, thanks a lot for the support.
